trieste/models/gpflux/interface.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+import tensorflow as tf
+from gpflux.models import DeepGP
+
+from ...data import Dataset
+from ...types import TensorType
+from ..gpflow.utils import module_deepcopy
+from ..interfaces import ProbabilisticModel
+from ..optimizer import TFOptimizer
+
+
+class GPfluxPredictor(ProbabilisticModel, tf.Module, ABC):
+    """A trainable wrapper for a GPflux deep Gaussian process model."""
+
+    def __init__(self, optimizer: TFOptimizer | None = None):
```

Comment:
ok, I would remove dependency on `TFOptimizer` altogether if we are relying on `fit` method, its essentially used for passing around arguments for fit and defining the loss function - instead we can pass the optimizer and fit arguments here in init and require models (subclasses) to implement loss method

Reply:
well actually `optimizer` is not needed at all here if we use fit method...